### DISCUSSION TITLE
Shows, Episodes, Hosts, & Guests

### DISCUSSION THREAD

**Discussion by rawkode** [RawkodeAcademy/RawkodeAcademy#310] [Discussions]
I likely took a slightly naive approach to this on the first pass, but now I want to make sure we get something correct that works long term.

Currently, we have:

- Video
- Shows
- Episodes
- People
- Show Hosts

Where:

- Video is a single piece of video content
- A Show is a Show
- An Episode is a Video attached to a Show with a unique Episode Code
- People are people
- Show Hosts are People attached to Shows

Now I want to add Episode Guests.

I started to implement this with a service called `episode-guests` like we did for `show-hosts`.

However, it made me think.

- What about guests on videos that aren't attached to shows?
- What if a Show has a temporary Host for one, or more, episodes?

Should we actually be having a cast list for each video, where each person has a "role"?

**Comment by rawkode** [comment]
Show Hosts should still exist, as it documents the "current" host of any show; and makes displaying show information trivial.

However, what if we change the host over time? It may not always be me. Allocating roles to videos would make this a solid improvement, but I do understand I'm preparing for circumstances that may not exist for many months, year, or perhaps never.

**Comment by swade1987** [comment]
Here's a proposed approach that provides both current functionality and future flexibility:

Instead of creating a separate `episode-guests` service, you should consider a more generic `video-cast` service that handles all participants in any video (both show episodes and standalone videos).

Each `VideoCast` entry would represent one person's role in one video. For example, a video with one host and two guests would have 3 VideoCast entries:

```
VideoCast {
  videoId: string;
  personId: string;
  role: string;  // e.g. "HOST", "GUEST", "CO_HOST"
  startTimestamp?: Date;
  endTimestamp?: Date;
  notes?: string;
}
```

This approach:

- Handles both show episodes and standalone videos
- Maintains historical accuracy for role changes over time
- Supports temporary roles (like guest hosts)
- Keeps entries atomic for easy querying and updating

The existing show-hosts service should remain as it serves the specific purpose of documenting current show hosts, making show information displays straightforward.

This structure gives flexibility for future scenarios like:

- Guest appearances on non-show videos
- Temporary/rotating hosts
- Multiple hosts/co-hosts
- Historical tracking of role changes
